src/schematools/contrib/django/management/commands/import_schemas.py

```diff
@@ -62,16 +63,50 @@ def import_from_url(self, schema_url) -> List[Dataset]:
 
     def import_schema(self, name: str, schema: DatasetSchema) -> Optional[Dataset]:
         """Import a single dataset schema."""
+
+        created = False
         try:
-            dataset = Dataset.objects.get(name=to_snake_case(schema.id))
+            dataset = Dataset.objects.get(name=Dataset.name_from_schema(schema))
```

Comment:
Perhaps `Dataset.get_schema_name()` as classmethod name?

Reply:
It is not really schema name, but rather Dataset name generator based on schema definition.